scripts/differential_check.py

```
@@ -284,22 +343,80 @@ def _hash(array) -> str:
+    # Hashed before anything runs, so a call which writes into its own
+    # argument can be told from one which does not. Nothing else here
+    # would notice: every digest is taken from the call's own result.
+    inputs_before = _input_digests(calls)
+    out, timing = {}, {}
+    for name, call in calls.items():
         try:
-            out[name] = digest(call())
+            seconds, patch = _timed(call)
+            out[name] = digest(patch)
+            timing[name] = seconds
         except Exception as error:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)